samples/web/content/apprtc/gcm_register.py

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/python2.4
+#
+# Copyright 2015 Google Inc. All Rights Reserved.
+
+"""AppRTC GCM ID Registration
+
+This module implements user ID registration and verification.
+"""
+
+import collections
+import constants
+import datetime
+import json
+import logging
+import webapp2
+
+from google.appengine.ext import ndb
+
+def get_ancestor_key():
+  return ndb.Key('GCMRecord', 'global')
+
+PARAM_USER_ID = 'userId'
+PARAM_GCM_ID = 'gcmId'
+PARAM_CODE = 'code'
+PARAM_OLD_GCM_ID = 'oldGcmId'
+PARAM_NEW_GCM_ID = 'newGcmId'
+PARAM_USER_ID_LIST = 'userIdList'
+
+def has_msg_field(msg, field):
+  return msg and field in msg and \
+      (not isinstance(msg[field], collections.Iterable) or \
+       len(msg[field]) > 0)
+
+def has_msg_fields(msg, fields):
+  return reduce(lambda x, y: x and y,
+                map(lambda x: has_msg_field(msg, x), fields))
+
+class BindPage(webapp2.RequestHandler):
+  def get_message_from_json(self):
+    try:
+      message = json.loads(self.request.body)
+      if isinstance(message, dict):
+        return message
+      logging.warning('Expected dictionary message'
+          + ', request=' + self.request.body)
+      return None
+    except Exception as e:
+      logging.warning('JSON load error from BindPage: error=' + str(e)
+          + ', request=' + self.request.body)
+      return None
+
```

Comment:
Function comments of the form
"Adds a new registration for a user ID and GCM ID, with a newly generated verification code."